### PR TITLE
As a user, can log in to the app via Google #184651333

### PR DESCRIPTION
**Pull request by kagame8732**
#### What does this PR do?
Login to the app via Google

#### Description of Task to be completed?
-Go to `http://localhost:3000/`
-Click on the "Authenticate with google" button, and you will be redirected to Google's login page. If you're already logged in to Google on your device, you may be taken directly to the authorization page
-On the Google login page, you'll be asked to enter your Google account email address and password. Once you've entered your login credentials, click "Next" to continue.
-After that, you may be asked to authorize the app to access your Google account information. This is a security measure to protect your privacy and ensure that you're comfortable with the website accessing your Google account data.
-If you're comfortable with the app accessing your Google account information, click "Allow". Once you've authorized the website, you should be redirected back to the website's login page, where you should now be logged in via Google.

-As a user, I should be able to log in via Google

-  Install the necessary NPM packages for google auth. 
   `npm install passport`
` npm install passport-google-oauth2`
   `npm install bcrypt `
   `npm install express-session `

- Set up your Node.js/Express application

#### How should this be manually tested?
- Clone the repository, 
- Navigate to the project branch,
- Open your favorite editors,
- In the terminal type: `npm install` to install the required dependencies.
-start the server with: `npm start`
-Navigate to: http://localhost:3000/
-Click on the button below the URL to redirect to the login page 

#### What are the relevant pivotal tracker/Trello stories?
#184651333

### CODE REVIEW

File: index.js
Comment (nshutip) [inline index.js:1]: Why were lines 37 - 44 deleted?

**Review by nshutip — COMMENTED** [review]
- The PR name does not meet the engineering playbook standards.
- You have to squash the commits into one.
- Fix the coveralls error.

**Review by ruberwa — COMMENTED** [review]
- Your test coverage are bellow minimum

File: index.js
Comment (kagame8732) [reply index.js:1]: Because according to what I rebase from develop branch  the structure of swagger was changed and deleted in order to remove data redundancy

File: src/controllers/user.controller.js [hunk 1]
```diff
@@ -20,23 +20,22 @@ export const createUser = async (req, res) => {
...
     email: req.body.email,
-    password: hashedPassword
+    password: bcrypt.hashSync(req.body.password, 10), // Hash the password
   };
 
+  // Save User in the database
   User.create(user)
     .then((data) => {
-      const token = jwt.sign({ email: data.email, id: data.id }, process.env.USER_SCREET_KEY);
-      res.status(200).json({ message: 'successful signedup', token });
```
Comment (JulesNIYOMWUNGERI) [inline src/controllers/user.controller.js]: why did you delete this

File: src/controllers/user.controller.js [hunk 2]
```diff
@@ -20,23 +20,22 @@ export const createUser = async (req, res) => {
     return res.send(error.details);
   }
 
-  const hashedPassword = await bcrypt.hash(req.body.password, 12);
-
+  // Create a User
   const user = {
-    fullname: `${req.body.firstname} ${req.body.lastname}`,
```
Comment (JulesNIYOMWUNGERI) [inline src/controllers/user.controller.js:26]: why did you change this?,  it have distroyed the validation

File: src/controllers/user.controller.js [hunk 3]
```diff
@@ -20,23 +20,22 @@ export const createUser = async (req, res) => {
...
+    names: req.body.names,
     email: req.body.email,
-    password: hashedPassword
```
Comment (JulesNIYOMWUNGERI) [inline src/controllers/user.controller.js]: this also should not be changed,  and the whole signup controller should not be changed, as it will destroy validation as well as login functionality, because login have been implemented based on signup

File: src/models/user.js [hunk 4]
```diff
@@ -4,30 +4,15 @@ const {
...
-    },
-    email: {
-      type: DataTypes.STRING,
-      allowNull: false,
-      unique: true,
-    },
-    password: {
-      type: DataTypes.STRING,
-      allowNull: false
-    }
```
Comment (JulesNIYOMWUNGERI) [inline src/models/user.js]: why did you change this?

File: src/tests/app.test.js [hunk 5]
```diff
@@ -20,60 +20,68 @@ describe('Sample Test', () => {
...
-
-    const res = await chai.request(app)
-      .post('/requestPasswordReset')
-      .send({ email: 'ruberwa3@gmail.com' });
-
-    expect(res).to.have.status(200);
-    expect(res.body).to.have.property('message', 'Password reset email sent');
-
-    const resetToken = await db.ResetToken.findOne({ where: { userId: user.id } });
-    expect(resetToken).to.exist;
```
Comment (JulesNIYOMWUNGERI) [inline src/tests/app.test.js]: why deleting these tests

File: src/tests/app.test.js [hunk 6]
```diff
@@ -20,60 +20,68 @@ describe('Sample Test', () => {
...
   });
-
-  // CREATE USER
-  it('should register user and return a response with status code 200', (done) => {
-    const User = {
-      firstname: 'ABC',
-      lastname: 'ABC',
-      email: 'abc@gmail.com',
-      password: 'ABC'
-    };
```
Comment (JulesNIYOMWUNGERI) [inline src/tests/app.test.js]: do not delete this tests

**Review by JulesNIYOMWUNGERI — COMMENTED** [review]
don't change the signup structure and do not delete other tests

File: src/routes/routes.js
Comment (Raphaela664) [inline src/routes/routes.js:1]: What is this file for?

**Review by Raphaela664 — COMMENTED** [review]
Some files and folders seem to be missing eg: signup files, migration folder

**Review by nshutip — DISMISSED** [review]
Task well implemented but your branch has a conflict that needs to be resolved!

**Review by bahati10 — APPROVED** [review]
Please consider squashing your commits

**Review by heshima18 — APPROVED** [review]

**Review by bahati10 — APPROVED** [review]
Well done for the task
